File: function-app/Domain/DiagProcess.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DiagnosticExplorer.Domain;

namespace DiagnosticExplorer.Api.Domain;

public class DiagProcess
{
    public int Id { get; set; }
    public int SiteId { get; set; }
    public string? InstanceId { get; set; }
    public string Name { get; set; } = "";
    public string? UserName { get; set; } = "";
    public string? MachineName { get; set; } = "";

    public DateTime LastOnline { get; set; }
    public bool IsOnline { get; set; }
    public bool IsSending { get; set; }
    
    public DateTime LastReceived { get; set; }
}

File: angular-app/src/app/members/diagnostics/Model/DiagProcess.ts
import 'reflect-metadata';
import {signal} from "@angular/core";
import {DiagProcess} from "@domain/DiagProcess";


export class DiagProcessSignals {
  id = signal('');
  siteId = signal('');
  instanceId = signal('');
  processName = signal('');
  userName = signal('');
  lastOnline = signal(new Date());
  isOnline = signal(false);
  machineName = signal('');

  constructor(process?: DiagProcess) {
    if (process) {
      this.update(process);
    }
  }

  update(process: DiagProcess) {
    this.id.set(process.id);
    this.siteId.set(process.siteId);
    this.instanceId.set(process.instanceId);
    this.processName.set(process.processName);
    this.userName.set(process.userName);
    this.lastOnline.set(new Date(process.lastOnline));
    this.isOnline.set(process.isOnline);
    this.machineName.set(process.machineName);
  }
}

